secure_storage_manager.go

```diff
@@ -190,30 +190,30 @@ func readTemporaryCredential(sc *snowflakeConn, credType string) string {
 // Writes to temporary credential file when OS is Linux.
 func writeTemporaryCredential(sc *snowflakeConn, credType, token string) {
 	target := convertTarget(sc.cfg.Host, sc.cfg.User, credType)
+	credCacheLock.Lock()
+	defer credCacheLock.Unlock()
 	localCredCache[target] = token
 
 	j, err := json.Marshal(localCredCache)
 	if err != nil {
 		logger.Debugf("failed to convert credential to JSON.")
 	}
-	temporaryCredCacheLock.Lock()
 	writeTemporaryCacheFile(j)
-	temporaryCredCacheLock.Unlock()
 }
 
 func deleteTemporaryCredential(sc *snowflakeConn, credType string) {
 	if credCacheDir == "" {
 		logger.Debug("Cache file doesn't exist. Skipping deleting credential file.")
 	} else {
+		credCacheLock.Lock()
+		defer credCacheLock.Unlock()
 		target := convertTarget(sc.cfg.Host, sc.cfg.User, credType)
 		delete(localCredCache, target)
 		j, err := json.Marshal(localCredCache)
 		if err != nil {
 			logger.Debugf("failed to convert credential to JSON.")
```

Comment:
Same here. Also, maybe we can log as warn?

Reply:
addd return to the error scenario + logging it as Warn instead of Debug